src/main/io/vtx_smartaudio.c

```diff
@@ -328,17 +326,22 @@ static void saProcessResponse(uint8_t *buf, int len)
              vtxSmartAudio.capability.powerCount, vtxSmartAudio.powerValues[0], vtxSmartAudio.powerValues[1],
              vtxSmartAudio.powerValues[2], vtxSmartAudio.powerValues[3]));
             //dprintf(("processResponse: V2.1 received vtx power value %d\r\n",buf[7]));
-
-            saDevice.power = 0;//set to unknown power level if the reported one doesnt match any of the known ones
-            for (int8_t i = 0; i < vtxSmartAudio.capability.powerCount; i++) {
-                if(buf[7] == vtxSmartAudio.powerValues[i]) {
+            rawpowervalue = buf[7];
+        }
 #ifdef USE_SMARTAUDIO_DPRINTF
-                    if (saDevice.power != i + 1) {
-                        dprintf(("processResponse: V2.1 power changed from index %d to index %d\r\n", saDevice.power, i + 1));
-                    }
+        int8_t prevpower = saDevice.power;
```

Comment:
Needs to be made conditional to `USE_SMARTAUDIO_DPRINTF`. (Also, the debug logging in this file is getting very messy and will probably have to be removed at some point.)

Reply:
It is conditional to `USE_SMARTAUDIO_DPRINTF`. The old code also was, which is why it doesn't show up in the diff.